### PR TITLE
Front end

### PR DESCRIPTION
**Pull request by axcochrane**
Please see first set of front-end changes. Created layout template.

Thanks!
S&P

### CODE REVIEW

**Comment by JessicaBarclay** [comment]
:shipit:

Awesome! Nice nav

**Review by anaalta — APPROVED** [review]

**Comment by anaalta** [comment]
:shipit:

**Comment by adc17** [comment]
A couple of points:
—Look into using partials.
—What is all this `welcome` stuff?

**Review by AlexChalk — CHANGES_REQUESTED** [review]
A couple of requests:
—DRY out code with partials (see this page: http://guides.rubyonrails.org/layouts_and_rendering.html).
—Explain what all the welcome stuff is in your pr message (ideally in your commit messages next time).

**Comment by chrisjmit** [comment]
:shipit:

**Comment by axcochrane** [comment]
Thanks guys. We've made a couple of changes:

1 - abstracted _header and _footer partials from application layout template. 
2 - Create a new button class in our CSS to style the 'new post' button

Re the Welcome controller and views, I believe that's just the standard rails terminology for what we have previously referred to as index. So in essence, we've just created a path and a view for the / route.

Thanks!
S&P

File: app/assets/stylesheets/application.css [hunk 1]
```diff
@@ -13,3 +13,5 @@
  *= require_tree .
  *= require_self
  */
+
+require_relative styles.css
```
Comment (dbugsy) [inline app/assets/stylesheets/application.css]: This is not needed - your styles.css will be required and compiled automatically by the line `= require_tree .`.

**Comment by AlexChalk** [comment]
Hey, would definitely put stuff like `DOCTYPE` tags into partials. 

I think `/` is just defined on line 6 of your `config/routes` (the `root` line). But if `welcome` is the standard name for the stuff used to serve that page, and it's a RESTful name, etc., that's fine.

File: spec/helpers/welcome_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# describe WelcomeHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe WelcomeHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (dbugsy) [inline spec/helpers/welcome_helper_spec.rb]: looks like an unnecessarily generated file - check for these and delete them before merging.

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1,7 +1,15 @@
...
+
+<body>
+
 <% @posts.each do |post| %>
   <p><%= post.message %></p>
 <% end %>
 
 <%= link_to new_post_path do %>
-  New post
+  <button class='button'>New Post</button>
```
Comment (dbugsy) [inline app/views/posts/index.html.erb]: http://api.rubyonrails.org/classes/ActionView/Helpers/UrlHelper.html#method-i-button_to

**Review by dbugsy — CHANGES_REQUESTED** [review]
A couple of un-railsy things in here, otherwise awesome!

Be careful including images in the main directory - they go into the git repo hang around even after they are deleted. Better to find an external repository.

**Comment by dbugsy** [comment]
I think there is a failing test in this PR. Make sure to find it and fix it before merging.

**Comment by Robert-G-J** [comment]
Is the test Dan mentioned still failing @axcochrane ?

**Comment by FiddlersCode** [comment]
@Robert-G-J tests all passing, FINALLY :)

**Comment by dbugsy** [comment]
:shipit: :shipit:

**Pull request merged**